### DISCUSSION TITLE
Any way to divide movement into sections?

### DISCUSSION THREAD

**Discussion by cbradburne** [luni64/TeensyStep4#21] [Ideas]
Hi, I wanted to use this on a camera to create timelapse videos. If I set a start and stop point, with the image capture triggered by a switch, I was wondering if there's a way to divide that movement by a number of "steps" / number of photos I want to take to get to the final position.

I've got all the storage and movement sorted, but not the 'dividing the moves into small moves' bit. I wasn't sure if there was a way to say how many clock pulses to advance or something.

**Comment by luni64** [comment]
Maybe I don't understand correctly, but if you want to take pictures at certain positions (while the motors move), you could simply monitor the current position and trigger the camera if  you are at the correct position.  I'd probably do the monitoring in a timer interrupt.

**Reply by cbradburne** [reply]
It's to capture a 3D printer printing layer by layer. I have a switch attached to part of the printer where the head moves in-between layers and it currently works to take static time-lapses.

I would like to use the slider I made to make them look more dynamic. I would like the group move to only move a small part of the say to the final position every time the switch is pushed. The whole move divided by the number of layers of the 3D print.

I'm probably using the wrong words, but I'm assuming TeensyStep uses timers, if the whole move uses 1,000,000 timer steps, I could divide the move in to number of layers, and tell TeensyStep to only move 1,000,000 / number of layers each time the button is pressed.

Not sure I've explained that well at all.

Essentially, divide a move in to a number of small steps, and move to the next step on a button push. I only want to record the start and stop position, not each small step between.

**Reply by cbradburne** [reply]
It'll probably help to say the interval for each layer isn't the same so a set time between moves isn't possible, hence the switch to take a photo and move.

**Reply by luni64** [reply]
Let me make sure I understand correctly:

- You have a 3D printer (not controlled by TeensyStep), and you want to use a camera slider (controlled by TeensyStep) to create a time-lapse video of the printing process.
- Right now, your camera (static) is triggered by a switch that detects when the printer completes a layer, and that part works fine.
- Now, you want the camera slider to move gradually toward the final position while the printer is printing. Specifically, you want to divide the total slider movement into **small steps**, with each step triggered by the switch when a new layer starts.

If that’s correct, here’s how I would do it
1. Divide the total slider movement (i.e. the targets of the motors) by the number of layers in your print. 
2. Each time  the layer switch is triggered, increase the motor targets by these values and start the movement

Does that make sense?  I need to leave now, but I can answer tomorrow if this didn't help.

**Reply by cbradburne** [reply]
Yeah, true. I could probably do it in code. I'll give it a go and let you know if I'm missing something.
Thank you for taking the time to write.
